spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -160,6 +172,32 @@ public RewriteManifestsSparkAction stagingLocation(String newStagingLocation) {
     return this;
   }
 
+  @Override
+  public RewriteManifestsSparkAction clusterBy(List<String> clusteringColumns) {
+    // Collect set of allowable partition columns to cluster on
+    Set<String> availablePartitionNames =
+        spec.fields().stream().map(PartitionField::name).collect(Collectors.toSet());
+
+    // Check if these partition fields are included in the spec
+    Preconditions.checkArgument(
+        clusteringColumns.stream().allMatch(availablePartitionNames::contains),
+        "Cannot use custom clustering to rewrite manifests '%s'. All partition columns must be "
+            + "defined in the current partition spec: %s. Choose from the available partitionable columns: %s",
+        clusteringColumns,
+        this.spec.specId(),
```

Comment:
Why should the spec ID and not the spec?

I think this should probably be something like
```
"Cannot set manifest clustering because specified  field(s) %s were not found in current partition spec %s.",
missingFields,
this.spec
```

Reply:
Can change